File: BE/src/main/java/ma/uae/utils/ConsulterStudent.java
package ma.uae.utils;

import java.util.Date;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ConsulterStudent {
	private Long idstudent;

	private String nom;
	private String prenom;
	private String lieuDeNaissance;
	private String paysDenaissance;
	private String nationalite;
	private String cNE;
	private Date dateDeNaissance;
	private String cIN;
	private String organismeEmployeur;
	private String adresseDeCorrespondance;
	private String telephone;
	private String email;
	private String etatCivil;
	private String professionDePere;
	private String professionDeMere;
	private String myProfession;
	private String sexe;

	private String bacspecialite;
	private String lycee;
	private String academie;
	private Date bacdateDObtention;
	private String bacmention;

	private String licencespecialite;
	private String licenceetablissement;
	private String licenceunivertiste;
	private String licencepays;
	private Date licencedateDObtention;
	private String licencemention;

	private String masterspecialite;
	private String masteretablissement;
	private String masterunivertiste;
	private String masterpays;
	private Date masterdateDObtention;
	private String mastermention;

	private String specialite;
	private String etablissement;
	private String pays;
	private Date dateDObtention;
	private String mention;

	private String langue1;
	private String niveau1;
	private String langue2;
	private String niveau2;
	private String langue3;
	private String niveau3;
	private String langue4;
	private String niveau4;

	private String etablissement1;
	private String fonction1;
	private Date periodeCommence1;
	private Date periodeFin1;
	private String etablissement2;
	private String fonction2;
	private Date periodeCommence2;
	private Date periodeFin2;
	private String etablissement3;
	private String fonction3;
	private Date periodeCommence3;
	private Date periodeFin3;

	private String ced;
	
	private CopiesDto copies;

}

File: FE/src/app/model/ConsulterStudent.ts
import { copiesDto } from "./copiesDto";

export class Consulter {
    idstudent = 0;
    nom:string = "";
    prenom:string = "";
    dateDeNaissance = Date();
    lieuDeNaissance:string = "";
    paysDenaissance :string = "";
    nationalite :string = "";
    professionDePere :string = "";
    professionDeMere :string = "";
    sexe :string = "";
    cNE :string = "";
    cIN :string = "";
    etatCivil :string = "";
    myProfession :string = "";
    organismeEmployeur:string = "";
    adresseDeCorrespondance :string = "";
    telephone :string = "";
    email:string = "";

    bacspecialite:string = "";
    bacdateDObtention = Date();
    lycee :string = "";
    academie :string = "";
    bacmention :string = "";

    licencespecialite :string = "";
    licencedateDObtention = Date();
    licenceetablissement :string = "";
    licenceunivertiste :string = "";
    licencepays :string = "";
    licencemention :string = "";

    masterspecialite :string = "";
    masterdateDObtention = Date();
    masteretablissement :string = "";
    masterunivertiste :string = "";
    masterpays :string = "";
    mastermention:string = "";

    specialite :string = "";
    dateDObtention = Date();
    etablissement :string = "";
    pays :string = "";
    mention :string = "";

    langue1 :string = "";
    niveau1 :string = "";
    langue2 :string = "";
    niveau2 :string = "";
    langue3 :string = "";
    niveau3 :string = "";
    langue4 :string = "";
    niveau4 :string = "";

    etablissement1 :string = "";
    fonction1 :string = "";
    periodeCommence1 = Date();
    periodeFin1 = Date();
    etablissement2 :string = "";
    fonction2 :string = "";
    periodeCommence2 =Date();
    periodeFin2 = Date();
    etablissement3 :string = "";
    fonction3:string = "";
    periodeCommence3 = Date();
    periodeFin3 = Date();

    ced :string = "";

    copies?:copiesDto;
  }
